provider/datagen/tests/verify-zero-copy.rs

```diff
@@ -2,107 +2,118 @@
+
+            for locale in
+                IterableDynamicDataProvider::<icu_provider::datagen::ExportMarker>::supported_locales_for_key(
+                    &locale_provider, key,
+                ).unwrap()
+            {
+                let payload = postcard_provider.load_buffer(
+                    key,
+                    DataRequest {
+                        locale: &locale,
+                        metadata: Default::default(),
+                    },
+                ).unwrap().take_payload().unwrap();
+
+                let stats_before = dhat::HeapStats::get();
+
+                // We need to generate the stats before the deserialized struct gets dropped, in order
+                // to distinguish between a temporary and permanent allocation.
+                let stats_after = icu_datagen::deserialize_and_discard(key, payload, dhat::HeapStats::get).unwrap();
+
+                let vio_total = stats_after.total_bytes - stats_before.total_bytes;
+                let vio_net = stats_after.curr_bytes - stats_before.curr_bytes;
+                max_total_violation = cmp::max(vio_total, max_total_violation);
+                max_net_violation = cmp::max(vio_net, max_net_violation);
+            }
+            if max_total_violation != 0 {
+                if max_net_violation != 0 {
+                    net_violations.insert(key.path().get());
+                } else {
+                    total_violations.insert(key.path().get());
+                }
             }
         }
+
+        assert!(total_violations.iter().eq(EXPECTED_TOTAL_VIOLATIONS.iter()) && net_violations.iter().eq(EXPECTED_NET_VIOLATIONS.iter()),
+            "Expected violations list does not match found violations!\n\
+            If the new list is smaller, please update EXPECTED_VIOLATIONS in verify-zero-copy.rs\n\
+            If it is bigger and that was unexpected, please make sure the key remains zero-copy, or ask ICU4X team members if it is okay\
+            to temporarily allow for this key to be allowlisted.\n\
+            Expected (net):\n{EXPECTED_NET_VIOLATIONS:?}\nFound (net):\n{net_violations:?}\nExpected (total):\n{EXPECTED_TOTAL_VIOLATIONS:?}\nFound (total):\n{total_violations:?}")
     }
+}
 
-    assert!(total_violations.iter().eq(EXPECTED_TOTAL_VIOLATIONS.iter()) && net_violations.iter().eq(EXPECTED_NET_VIOLATIONS.iter()),
-        "Expected violations list does not match found violations!\n\
-        If the new list is smaller, please update EXPECTED_VIOLATIONS in verify-zero-copy.rs\n\
-        If it is bigger and that was unexpected, please make sure the key remains zero-copy, or ask ICU4X team members if it is okay\
-        to temporarily allow for this key to be allowlisted.\n\
-        Expected (net):\n{EXPECTED_NET_VIOLATIONS:?}\nFound (net):\n{net_violations:?}\nExpected (total):\n{EXPECTED_TOTAL_VIOLATIONS:?}\nFound (total):\n{total_violations:?}")
+#[cfg(target_os = "windows")]
+#[test]
+fn test_zero_copy() {
+    // do nothing
 }
```

Comment:
Nit: Probably better to omit the test entirely than make a stub test that looks like it is passing

Reply:
I didn't want to have trouble with a `cargo test`  target that contains no tests. I think it works (and marks the test as "ignored"), but I didn't want to experiment, we can remove it in a followup